### PR TITLE
Get article/:article_id

### PR DESCRIPTION
**Pull request by hedieh623**
request journey:
​
```mermaid
graph TD
    A>Request] -->|App| B(app.js)
    B --> |Controller| C(Article.controller.js)
    C --> |Model| D(Article.model.js)
    D --> E(db/connections.js)
    E --> F[(Database)]
```

### CODE REVIEW

**Comment by hedieh623** [comment]
[4. GET /api/articles/:article_id](https://trello.com/c/YYniSLXQ/12-4-get-api-articles-articleid)

File: Controllers/Article.controller.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const selectArticle = require("../Models/Article.model.js");
+
+const getArticle = (req, res) => {
+  selectArticle(req.params.article_id).then((article) => {
+    if(article){
+        res.status(200).send(article);
```
Comment (yalicee) [inline Controllers/Article.controller.js]: Needs to be sent on the key of article

File: __tests__/app.test.js [hunk 2]
```diff
@@ -10,25 +11,57 @@ describe("GET /api/topics", () => {
...
-      .expect(200)
-      .then((res) => {
-        const firstElement = res.body[0];
-        expect(typeof firstElement.slug).toBe("string");
-        expect(typeof firstElement.description).toBe("string");
-      });
+    .get("/api/topics")
+    .expect(200)
+    .then((res)=>{
+      const firstElement = res.body[0]
```
Comment (yangalexyangg) [inline __tests__/app.test.js:20]: Needs to be on key of article

File: __tests__/app.test.js [hunk 3]
```diff
@@ -10,25 +11,57 @@ describe("GET /api/topics", () => {
...
+});
+
+
+describe('GET /api/articles/:article_id', () => {
+  test('should call this endopoint and sees if it is an object or not and if the article id matches the URL"', () => {
+    return request(app)
+    .get("/api/articles/1")
+    .expect(200)
+    .then((res)=>{
+      const article_id = res.body.article_id
```
Comment (yalicee) [inline __tests__/app.test.js]: Needs to be on key of article

File: __tests__/app.test.js [hunk 4]
```diff
@@ -10,25 +11,57 @@ describe("GET /api/topics", () => {
...
+      expect(typeof(res.body)).toBe("object")
+      expect(article_id).toBe(1)
+      expect(typeof res.body.author).toBe("string");
```
Comment (yalicee) [inline __tests__/app.test.js]: must be values related to the requested article_id, e.g. NOT author: expect.any(String)

**Review by yalicee — CHANGES_REQUESTED** [review]
See comments please. Otherwise ok :)

File: Controllers/Article.controller.js (hunk 1 above)
Comment (hedieh623) [reply Controllers/Article.controller.js]: Hi alex
im not sure if i get what you mean. Could you please clarify?

File: __tests__/app.test.js (hunk 3 above)
Comment (hedieh623) [reply __tests__/app.test.js]: im also not sure if i get what you mean with this. Could you please clarify?

File: __tests__/app.test.js (hunk 4 above)
Comment (hedieh623) [reply __tests__/app.test.js]: ah yeah thanks!  have added the relevant expects now.

File: Controllers/Article.controller.js (hunk 1 above)
Comment (Rayhaan2k9) [reply Controllers/Article.controller.js]: Hi Hedieh. As it stands, you are sending the article object itself back as a response but it needs to be within another object with a key of article. You can do this by wrapping `article` on line 7 in curly brackets so it looks like `{ article }`

File: __tests__/app.test.js [hunk 5]
```diff
@@ -56,3 +60,37 @@ describe("GET /api/topics", () => {
...
+      const created_at = res.body.created_at;
+      const votes = res.body.votes;
+
+      expect(typeof(res.body)).toBe("object")
+      expect(article_id).toBe(1)
+      expect(author).toBe("butter_bridge")
+      expect(title).toBe("Living in the shadow of a great man")
+      expect(topic).toBe("mitch")
+      expect(body).toBe("I find this existence challenging")
+      //expect(Date.parse(created_at)).toBe(1594329060000);
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js]: remove this commented code please :)

File: __tests__/app.test.js [hunk 6]
```diff
@@ -56,3 +60,37 @@ describe("GET /api/topics", () => {
...
+      expect(votes).toBe(100)
+    })
+  });
+
+  test("should return the response(an object) that has the stated properties ", () => {
+    return request(app)
+      .get("/api/articles/1000")
+      .expect(404);
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js]: We should also be sending an error message back here to go with the 404 status. Also you should add another test for a bad request for example `/api/articles/banana`

**Review by Rayhaan2k9 — CHANGES_REQUESTED** [review]
A couple things to fix here, Hedieh. If you are unsure about anything, please drop an nchelp in 🙂

**Pull request merged**